src/NuGet.Core/NuGet.Commands/SignCommand/CertificateProvider.cs

```diff
@@ -171,7 +171,24 @@ private static X509Certificate2Collection LoadCertificateFromStore(CertificateSo
 
             if (!string.IsNullOrEmpty(options.Fingerprint))
             {
-                resultCollection = store.Certificates.Find(X509FindType.FindByThumbprint, options.Fingerprint, validOnly);
+                if (options.FingerprintHashAlgorithmName == Common.HashAlgorithmName.SHA1)
+                {
+                    resultCollection = store.Certificates.Find(X509FindType.FindByThumbprint, options.Fingerprint, validOnly);
+                }
+                else if (options.FingerprintHashAlgorithmName != Common.HashAlgorithmName.Unknown)
+                {
+                    foreach (var cert in store.Certificates)
+                    {
+                        string actualFingerprint = CertificateUtility.GetHashString(cert, options.FingerprintHashAlgorithmName);
+
+                        if (string.Equals(actualFingerprint, options.Fingerprint, StringComparison.InvariantCultureIgnoreCase))
+                        {
+                            resultCollection.Add(cert);
+                            break;
+                        }
+                    }
```

Comment:
Learnt this approach to search for a certificate in https://github.com/dotnet/sign/blob/1df56d5c09b004fec6b5413dc6c61183553589ec/src/Sign.SignatureProviders.CertificateStore/CertificateStoreService.cs#L154-L177